server/gc/fsck.c

```diff
@@ -65,46 +116,72 @@ check_blocks (SeafFSManager *mgr, FsckRes *res, const char *file_id)
     return ret;
 }
 
-static gboolean
-fs_callback (SeafFSManager *mgr,
-             const char *store_id,
-             int version,
-             const char *obj_id,
-             int type,
-             void *user_data,
-             gboolean *stop)
+// checked dir must be exists
+static char*
+fsck_check_dir_recursive (SeafFSManager *mgr,
+                          const char *id,
+                          FsckData *fsck_data)
 {
-    FsckRes *res = user_data;
-
-    if (type == SEAF_METADATA_TYPE_FILE && check_blocks (mgr, res, obj_id) < 0)
-        return FALSE;
-
-    return TRUE;
-}
-
-static gboolean
-check_fs_integrity (SeafCommit *commit, void *vdata, gboolean *stop)
-{
-    FsckRes *res = vdata;
-
-    /* Stop traversing commits after finding the first consistent commit. */
-    if (res->consistent_head != NULL) {
-        *stop = TRUE;
-        return TRUE;
+    SeafDir *dir;
+    SeafDir *new_dir;
+    GList *p;
+    SeafDirent *seaf_dent;
+    char *dir_id;
+    char *repo_id = fsck_data->repo->store_id;
+    int version = fsck_data->repo->version;
+    gboolean is_corrupted = FALSE;
+
+    dir = seaf_fs_manager_get_seafdir (mgr, repo_id, version, id);
+
+    for (p = dir->entries; p; p = p->next) {
+        seaf_dent = p->data;
+
+        if (S_ISREG(seaf_dent->mode)) {
+            if (check_blocks (mgr, fsck_data, seaf_dent->id) < 0) {
+                // file curropted, set it empty
+                seaf_message ("File %s(%.8s) is curropted, set to empty.\n",
```

Comment:
You should print different error message when clean is false. In that case don't say you set something empty. It will make the user nervous.